airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/SyncWorkflowImpl.java

```diff
@@ -40,9 +53,32 @@ public StandardSyncOutput run(final JobRunConfig jobRunConfig,
                                 final StandardSyncInput syncInput,
                                 final UUID connectionId) {
 
-    StandardSyncOutput syncOutput = replicationActivity.replicate(jobRunConfig, sourceLauncherConfig, destinationLauncherConfig, syncInput);
-
     final int version = Workflow.getVersion(VERSION_LABEL, Workflow.DEFAULT_VERSION, CURRENT_VERSION);
+    final StandardCheckConnectionInput sourceConfiguration = new StandardCheckConnectionInput()
+        .withConnectionConfiguration(syncInput.getSourceConfiguration());
+    final StandardCheckConnectionInput destinationConfiguration =
+        new StandardCheckConnectionInput().withConnectionConfiguration(syncInput.getDestinationConfiguration());
+
+    if (version >= VERSION_INTRODUCING_CHECK_BEFORE_SYNC) {
```

Comment:
I think that this block should be in the `ConnectionManagerWorkflow`, it is not really a part of the sync itself but part of what we need to check before the sync. This is something to be discuss but I believe that the check should live in the Connection manager.

Reply:
That version check was introduced to solve https://github.com/airbytehq/airbyte/pull/12423#pullrequestreview-957997758.  

I *think* what is happening is that with these changes the SyncWorkflow now has a new number of actions that it will (/might) run, and any in-progress workflows at the time of the upgrade will have trouble if new activities appear.  The ConnectionManagerWorkflow didn't gain or loose any sub-activities, so it doesn't need to be versioned.

Reply:
Moved to ConnectionManagerWorkflow in https://github.com/airbytehq/airbyte/pull/12423/commits/f0efd637a36957b56e974846e7e2b7d1dd6dcb40